### PR TITLE
Fixing warnings after running static code analyser.

### PR DESCRIPTION
**Pull request by kushalsingh007**
-Changes to fix all the clarifyCalculation warnings.
-Fixes a part of #480

### CODE REVIEW

**Comment by kushalsingh007** [comment]
Fixes for issue #480

File: cpu/cortex-m0_common/include/core_cmFunc.h [hunk 1]
```diff
@@ -342,7 +342,7 @@ __attribute__( ( always_inline ) ) __STATIC_INLINE void __disable_irq(void)
  */
 __attribute__( ( always_inline ) ) __STATIC_INLINE uint32_t __get_CONTROL(void)
 {
-  uint32_t result;
+  uint32_t result = -1;
 
   __ASM volatile ("MRS %0, control" : "=r" (result) );
```
Comment (gebart) [inline cpu/cortex-m0_common/include/core_cmFunc.h]: False positive, if I am not mistaken: This ASM operation writes the `control` special purpose register value to the register represented by %0, i.e. the `result` variable, which means that this variable is always written before `return(result);`
Therefore, it is safe to suppress static code warnings about uninitialized variables in this (and the following) ASM wrapper inlines.

**Comment by gebart** [comment]
Just a heads up: `core_cmFunc.h` and some other headers in the `cpu` directory are supplied by ARM or the CPU vendors and may be replaced from time to time when new versions of CMSIS are released. There may be other files which are more worthwhile to spend your effort on fixing so you don't feel like you are wasting your time.

**Comment by kushalsingh007** [comment]
Thanks for the heads up @gebart . I'll remove those one's from the commit for now then.

**Comment by kushalsingh007** [comment]
@gebart : Apart from the  clarifyCalculation , out of the other one's
* uninitvar: due to core_cmFunc.h and so no use working on them . 
* unusedVariable: I had a look and I think this warning is also not useful.
* memleak : I think the error is because it thinks that memleak can be a null pointer , but that can't be the case as checked 3-4 lines above it. 

So , should I  supress the remaining one's now ?

**Comment by kushalsingh007** [comment]
Why do I get the needs squashing error during build ?
```Pull request needs squashing```

**Comment by OlegHahm** [comment]
> Why do I get the needs squashing error during build ?

Because your commits starts with `Fix...` instead of a module prefix.

**Comment by kushalsingh007** [comment]
Changed it for now (temporarily changed from ```Fix.. ``` ) , what exactly  are module prefixes ?

**Comment by OlegHahm** [comment]
> Does that mean that suppose if changes are in cpu/ part of the code , then i should add CPU: in the front ? 

Yes, that's correct.

**Comment by kushalsingh007** [comment]
@OlegHahm : What do you think about the other warnings , should I suppress them now ?
>* uninitvar: due to core_cmFunc.h and so no use working on them .
>* unusedVariable: I had a look and I think this warning is also not useful.
>* memleak : I think the error is because it thinks that memleak can be a null pointer , but that can't be the case as checked 3-4 lines above it.

**Comment by OlegHahm** [comment]
If you suppress them, please make an according comment.

Anyway I would vote for a separate PR for the rest of the fixes.

**Comment by OlegHahm** [comment]
btw. ACK for this PR as soon as Travis is fine.

**Comment by kushalsingh007** [comment]
Yes , travis is fine ```All is well```

**Pull request merged**